thetis/momentum_eq.py

```diff
@@ -714,4 +707,4 @@ def residual(self, solution, solution_old, fields, fields_old, bnd_conditions=No
                                   Dx(bhead, 1)*self.test[1])
             f = g_grav * grad_head_dot_test * self.dx
 
-        return f
+        return -f
```

Comment:
Ah ok, so `int_pg` is now -g grad r ? Still not sure what the logic is...

Reply:
Yeah so my idea was that `residual` should always return a term on the RHS. But what I did indeed flips the sign of the `int_pg` field which is counter-intuitive. I have now changed this so that all the terms are on RHS and `int_pg` has the expected sign.